src/main/kotlin/com/hitc/hitwplugin/main/Main.kt

```diff
@@ -1,18 +1,71 @@
 package com.hitc.hitwplugin.main
 
+import com.hitc.hitwplugin.main.commands.PasteWall
+import com.hitc.hitwplugin.main.core.BlockPalette
+import com.hitc.hitwplugin.main.core.WallBlockType
+import org.bukkit.DyeColor
+import org.bukkit.Material
+import org.bukkit.TreeSpecies
 import org.bukkit.event.EventHandler
 import org.bukkit.event.Listener
 import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.material.MaterialData
+import org.bukkit.material.Tree
 import org.bukkit.plugin.java.JavaPlugin
+import java.nio.file.Files
+import kotlin.collections.HashMap
+import kotlin.io.path.Path
 
 class Main: JavaPlugin(), Listener {
 
     override fun onEnable() {
         server.pluginManager.registerEvents(this, this)
+        registerCommands()
+        Files.createDirectories(Path("./plugins/HitW"))
     }
 
     @EventHandler
     fun onPlayerJoin(event: PlayerJoinEvent?) {
         event?.player?.sendMessage("Boop!")
     }
+
+    private fun registerCommands() {
+        getCommand("pastewall").executor = PasteWall()
+
+    }
+
+}
+
+object MainObject {
```

Comment:
`companion object` instead of `object MainObject` as you want to add an object to a class not define an object within it